### PR TITLE
Add Smart Contract for Automated Rewards & 10% Platform Fee

### LINKED ISSUE #24: Smart Contract Implementation: Automated Reward Distribution with 10% Platform Fee
### Is your feature request related to a problem? Please describe.

The current reward distribution system has several limitations that impact scalability, transparency, and user trust:

**Current Issues:**
1. **Security Risks**: Rewards are sent directly from pool wallets using private keys stored in the database, creating security vulnerabilities and single points of failure.
2. **Gas Fee Burden**: Factory creators must maintain ETH balances in their pools to pay for transaction fees, creating an ongoing financial burden.
3. **Lack of Transparency**: Users cannot verify reward calculations or platform fees independently.
4. **No Platform Revenue**: The current system doesn't automatically collect platform fees, requiring manual accounting.

**Proposed Solution:**
Implement a Base smart contract that allows farmers to withdraw their earned rewards directly, with automatic 10% platform fee collection. This shifts the gas fee burden from factory creators to farmers while ensuring transparent and automated fee collection.

### Describe the solution you'd like

**Smart Contract Architecture:**

1. **Reward Pool Contract**: A Solidity smart contract that manages reward distribution for all factories
2. **Withdrawal Function**: Farmers can call a `withdraw_rewards` function to claim their earned tokens
3. **Automatic Fee Collection**: 10% of each withdrawal is automatically sent to the platform treasury
4. **Multi-Token Support**: Support for $CLONES, $USDC, and other ERC20 tokens
5. **Batch Withdrawals**: Allow farmers to withdraw from multiple factories in a single transaction

**Key Features:**

- **Farmer-Initiated Withdrawals**: Farmers pay gas fees to withdraw their rewards
- **Platform Fee**: 10% of each withdrawal automatically goes to platform treasury
- **Transparent Calculations**: All reward calculations are verifiable on-chain
- **No Factory ETH Requirements**: Factory creators no longer need to maintain ETH balances
- **Audit Trail**: All withdrawals and fees are recorded on-chain

### Benefits

1. **Reduced Factory Creator Burden**: No more ETH balance requirements for gas fees
2. **Transparent Fee Collection**: 10% platform fee is automatically collected and verifiable
3. **Better User Experience**: Farmers control when to withdraw and pay gas fees
4. **Scalability**: Automated processing eliminates manual bottlenecks
5. **Audit Trail**: All transactions are recorded on-chain for transparency
6. **Cost Efficiency**: Farmers can batch withdrawals to minimize gas costs

### Security Considerations

1. **Smart Contract Auditing**: Full security audit before mainnet deployment
2. **Access Control**: Only authorized factory creators can record rewards
3. **Fee Validation**: Ensure platform fee calculations are accurate
4. **Emergency Pause**: Ability to pause withdrawals if issues are detected
5. **Upgrade Mechanism**: Plan for smart contract upgrades if needed

### Testing Requirements

- [x] Smart contract unit tests for all functions
- [x] Integration tests with Base Sepolia
- [x] Backend API tests for new endpoints
- [x] End-to-end testing of complete reward flow
- [x] Gas optimization testing for batch withdrawals
- [x] Security testing for edge cases

### Definition of Done

- [ ] #55
- [x] Backend processing service updated to record rewards instead of transferring
- [x] New API endpoints implemented and tested
- [x] Database schema updated with new reward tracking fields
- [x] Frontend updated to support farmer-initiated withdrawals
- [x] Migration completed for existing pools
- [x] Documentation updated for new reward system
- [x] Monitoring and alerting in place for smart contract events

### Additional Context

This implementation aligns with the new Factory/Gym terminology and provides a more sustainable and scalable approach to reward distribution. The smart contract will be a critical component of the platform's infrastructure and should be developed with production-grade security and reliability standards.

### PR DESCRIPTION
**Pull request by diamondly777**
Implemented smart contract reward system, security features and monitoring

### CODE REVIEW

File: src/services/forge/processing.ts [hunk 1]
```diff
@@ -15,15 +14,26 @@ import * as path from 'path';
...
 
+// Initialize reward pool service
+const rewardPoolService = RewardPoolService.getInstance(
+  new (await import('../blockchain/index.js')).default(
+    process.env.RPC_URL || '',
+    ''
+  ).connection,
+  process.env.REWARD_POOL_PROGRAM_ID || '11111111111111111111111111111111',
+  Keypair.generate() // TODO: Replace with actual platform authority keypair
+);
```
[the remark this answers is not in the document]
Comment (diamondly777) [reply src/services/forge/processing.ts]: Fixed Top-Level Await Import Issue
Implemented lazy initialization pattern with getRewardPoolService() function

[the remark this answers is not in the document]
Comment (diamondly777) [reply src/services/forge/processing.ts]: image

File: src/services/forge/processing.ts [hunk 2]
```diff
@@ -15,15 +14,26 @@ import * as path from 'path';
...
 const processingQueue: string[] = [];
 
+// Initialize reward pool service
+const rewardPoolService = RewardPoolService.getInstance(
+  new (await import('../blockchain/index.js')).default(
+    process.env.RPC_URL || '',
+    ''
+  ).connection,
+  process.env.REWARD_POOL_PROGRAM_ID || '11111111111111111111111111111111',
+  Keypair.generate() // TODO: Replace with actual platform authority keypair
```
[the remark this answers is not in the document]
Comment (diamondly777) [reply src/services/forge/processing.ts]: image
Added proper keypair loading from environment variable with error handling

File: src/services/forge/processing.ts [hunk 3]
```diff
@@ -459,13 +402,18 @@ export async function processNextInQueue() {
       submission.reward = reward;
       submission.maxReward = maxReward;
       submission.clampedScore = clampedScore;
-      submission.treasuryTransfer = treasuryTransfer;
+      
+      // Add smart contract reward data if available
```
[the remark this answers is not in the document]
Comment (diamondly777) [reply src/services/forge/processing.ts:422]: Removed the problematic reference and added explanatory comments about the expected behavior

File: src/services/monitoring/monitoringService.ts [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+  private isRelevantTransaction(transaction: ParsedTransactionWithMeta): boolean {
+    if (!transaction.meta || !transaction.transaction.message.instructions) {
+      return false;
+    }
+
+    // Check if any instruction involves our program
+    return transaction.transaction.message.instructions.some(
+      (instruction: ParsedInstruction) => {
+        if ('programId' in instruction) {
+          return instruction.programId.equals(this.programId);
```
[the remark this answers is not in the document]
Comment (diamondly777) [reply src/services/monitoring/monitoringService.ts]: image
Added proper type union and handling for both parsed and raw instructions

File: src/solana-client.ts [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+}
+
+// Deserialization helpers (these would need to be implemented based on your serialization format)
+// For now, using placeholder implementations
+export namespace RewardPool {
+  export function deserialize(data: Buffer): RewardPool {
+    // Implementation would depend on your serialization format
+    // This is a placeholder
+    return {
+      isInitialized: true,
```
[the remark this answers is not in the document]
Comment (diamondly777) [reply src/solana-client.ts]: image
Added proper buffer parsing logic with detailed layout comments
